### PR TITLE
CON-2878-Markdown-Breakdown-URL-Exercise

### PR DESCRIPTION
**Pull request by zanninso**
[CON-2878](https://01talent.atlassian.net/browse/CON-2878)

[CON-2878]: https://01talent.atlassian.net/browse/CON-2878?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: subjects/java/checkpoints/breakdown-url/README.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+## URL Parsing
+
+### Instructions
+
+Create a class `UrlParser` that provides a method to parse and validate URLs using regex. The method should extract and return url components the `protocol`, `domain`, `port`, `path` and `query` parameters. The URL is always correct.
```
Comment (nprimo) [inline subjects/java/checkpoints/breakdown-url/README.md:5]: url should be all in capital letters to keep the text consistent

File: subjects/java/checkpoints/breakdown-url/README.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+> give back in the map just the existing component.
```
Comment (nprimo) [inline subjects/java/checkpoints/breakdown-url/README.md:7]: `give` should be `Give` since it's the beginning of the sentece

File: subjects/java/checkpoints/breakdown-url/README.md (hunk 1 above)
Comment (nprimo) [inline subjects/java/checkpoints/breakdown-url/README.md:5]: I would rephrase the second phrase like:
```
The method should extract and return the following URL components: ....
Assume the URL is always correct.
```

**Review by nprimo — CHANGES_REQUESTED** [review]

File: subjects/java/checkpoints/breakdown-url/README.md (hunk 1 above)
Comment (zanninso) [reply subjects/java/checkpoints/breakdown-url/README.md:5]: done

File: subjects/java/checkpoints/breakdown-url/README.md (hunk 2 above)
Comment (zanninso) [reply subjects/java/checkpoints/breakdown-url/README.md:7]: done

File: subjects/java/checkpoints/breakdown-url/README.md (hunk 1 above)
Comment (zanninso) [reply subjects/java/checkpoints/breakdown-url/README.md:5]: yeah that's better, thanks for the suggestion.

**Review by nprimo — APPROVED** [review]
LGTM
